src/vit-testing/mainnet-tools/src/snapshot_wormhole/command.rs

```diff
@@ -0,0 +1,199 @@
+
+        let config = read_config(self.config)?;
+
+        match self.cmd {
+            Operation::OneShot => one_shot(&config),
+            Operation::Schedule(schedule) => schedule.exec(&config),
+        }
+    }
+}
+
+/// Performs one-time snapshot transport operation.
+///
+/// # Errors
+///
+/// On any errors from services
+#[instrument(fields(
+    source=config.snapshot_service.address.to_string(),
+    target=config.servicing_station.address.to_string(),
+    tag=config.parameters.tag
+   ),
+   skip(config)
+)]
+pub fn one_shot(config: &Config) -> Result<(), eyre::Report> {
+    info!("Job started");
+
+    let snapshot_params = JobParameters {
+        slot_no: None,
+        tag: Some(config.parameters.tag.clone()),
+    };
+
+    let snapshot_client = if let Some(token) = &config.snapshot_service.token {
+        debug!("Using token for snapshot retrieval");
+        SnapshotRestClient::new_with_token(
+            token.to_string(),
+            config.snapshot_service.address.to_string(),
+        )
+    } else {
+        debug!("no token for snapshot retrieval");
+        SnapshotRestClient::new(config.snapshot_service.address.to_string())
+    };
+
+    let snapshot_job_id = snapshot_client.job_new(snapshot_params.clone())?;
+
+    let wait = WaitBuilder::new().tries(10).sleep_between_tries(10).build();
+
+    debug!("Awaiting snapshot with strategy: {wait}");
+    let snapshot_jobs_status =
+        snapshot_client.wait_for_job_finish(snapshot_job_id.clone(), wait)?;
+
+    debug!("Snapshot done with status: {snapshot_jobs_status:?}");
+    let snapshot_content =
+        snapshot_client.get_snapshot(snapshot_job_id, snapshot_params.tag.unwrap_or_default())?;
+
+    let snapshot: Vec<Output> = serde_json::from_str(&snapshot_content)?;
+    let result = SnapshotResult::from_outputs(snapshot_jobs_status, snapshot)?;
+    debug!("snapshot parsed.");
+
+    let raw_snapshot_input = RawSnapshotRequest {
+        tag: config.parameters.tag.clone(),
+        content: RawSnapshotInput {
+            snapshot: RawSnapshot::from(result.registrations().clone()),
+            update_timestamp: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards")
+                .as_secs()
+                .try_into()?,
+            min_stake_threshold: config.parameters.min_stake_threshold,
+            voting_power_cap: config.parameters.voting_power_cap,
+            direct_voters_group: config.parameters.direct_voters_group.clone(),
+            representatives_group: config.parameters.representatives_group.clone(),
+        },
+    };
+
+    debug!("updating vit servicing station server...");
+    let vit_ss_rest_client = RestClient::new(config.servicing_station.address.parse()?);
+    vit_ss_rest_client.put_raw_snapshot(&raw_snapshot_input)?;
+    info!("transfer done.");
+    Ok(())
+}
+
+pub fn read_config<P: AsRef<Path>>(config: P) -> Result<Config, eyre::Report> {
```

Comment:
Looks like we're already importing `Result` from `color_eyre`, so I think this can just be `Result<Config>`